apps/api/src/integration/integration.service.ts

```diff
@@ -782,4 +778,96 @@ export class IntegrationService {
         break
     }
   }
+
+  /**
+   * Tests a new integration in the given workspace. The user needs to have
+   * `CREATE_INTEGRATION` and `READ_WORKSPACE` authority in the workspace.
+   * Validate only metadata and event subscriptions for an integration.
+   *
+   * This method skips project and environment resolution entirely, focusing on
+   * permitted events, metadata parameter validation and live configuration tests.
+   *
+   * @param user - The authenticated user performing metadata validation.
+   * @param dto - CreateIntegration or UpdateIntegration DTO containing optional
+   *   notifyOn and metadata fields.
+   * @param isCreate - True if validating for a new integration; false for update.
+   * @param integrationSlug - Slug of the existing integration (required when isCreate is false).
+   * @returns A promise resolving to { success: true } upon successful validation.
+   * @throws BadRequestException if integrationSlug is missing when updating.
+   * @throws UnauthorizedException if the user is not authorized to update the integration.
+   * @throws BadRequestException if event subscriptions or metadata parameters are invalid.
+   * @throws BadRequestException if live configuration testing via validateConfiguration fails.
+   */
+  async validateIntegrationMetadata(
+    user: AuthenticatedUser,
+    dto: CreateIntegration | UpdateIntegration,
+    isCreate: boolean,
+    integrationSlug?: Integration['slug']
+  ): Promise<{ success: true }> {
+    this.logger.log(
+      `User ${user.id} is metadata‐validating integration ${dto.name} ` +
+        (isCreate ? `(new)` : `(existing ${integrationSlug})`)
+    )
+
+    let integrationObject: BaseIntegration
+    if (isCreate) {
+      integrationObject = IntegrationFactory.createIntegrationWithType(
+        (dto as CreateIntegration).type,
+        this.prisma
+      )
+    } else {
+      if (!integrationSlug) {
+        throw new BadRequestException(

```

Comment:
This error should be an InternalError, since the users won't have any idea what an integration slug is, and why it's not present. 

We would throw the error like this: `constructErrorBody('Uh-oh, something went wront on our end', 'We have faced an issue while validating your integration. Please try again later, or get in touch with us at support@keyshade.xyz')`